### PR TITLE
Implement (x|y)other hovertemplate to format differing positions in compare and unified modes

### LINKED ISSUE #5555: Hovertemplate display of X value of points in hoverset that differ from that of the winning point
Right now, in compare and unified hover modes, it's possible for points to be in the hover set and still have slightly different X values from that of the point that won the hover, and in this case we show the differing X value in parentheses in the hoverlabel.

We need to be able to show this value when `hovertemplate` is active as well, possibly via some new hovertemplate %{value} that is empty if the X value matches that of the winning point, and contains "(%{x})" otherwise.

This should honor the new `(x|y)hoverformat` attributes too.

### PR DESCRIPTION
**Pull request by archmoj**
Resolves #5555.

TODOs:
- [x] add tests

@plotly/plotly_js

### CODE REVIEW

**Comment by archmoj** [comment]
How about renaming the key to `other(x|y)` or even simply `other`?

**Comment by nicolaskruchten** [comment]
I like `(x|y)other`... what's the impetus for the rename?

One challenge I see with such a conditionally-displaying item is the spacing around it... It's OK if I put this at the end of a hovertemplate like `%{y} %{xother}` because most of the time the trailing space won't be visible, but I can't as easily put it at the beginning necessarily as then the leading space will be visible: `%{xother} %{y}`... Can we imagine some formatting directives like `%{xother:<something>}` the way we do number-formatting?

**Comment by archmoj** [comment]
> I like `(x|y)other`... what's the impetus for the rename?
> 

We basically don't use `xother` and `yother` at the same time. 
So I thought there might be a potential to simplify the name.

**Comment by nicolaskruchten** [comment]
ah I see. I still think it's helpful to reason about and eventually find if it contains the letter :)

**Comment by archmoj** [comment]
> One challenge I see with such a conditionally-displaying item is the spacing around it... It's OK if I put this at the end of a hovertemplate like `%{y} %{xother}` because most of the time the trailing space won't be visible, but I can't as easily put it at the beginning necessarily as then the leading space will be visible: `%{xother} %{y}`... Can we imagine some formatting directives like `%{xother:<something>}` the way we do number-formatting?

The undesirable extra spacing won't be added after https://github.com/plotly/plotly.js/pull/5690/commits/ad2ad740e012eb43f623c041ce0a3204948b8a39.

**Comment by nicolaskruchten** [comment]
Yeah... but now you can't not-have the spacing around it if you don't want it, like if you want to have it at the beginning you'll always have a leading space when it's present. 

I have a slight preference for the way it was before, I was just trying to think of a way to do even better :)

File: src/lib/index.js [hunk 1]
```diff
@@ -1087,8 +1099,14 @@ function templateFormatString(string, labels, d3locale) {
                 value = lib.formatDate(ms, format.replace(TEMPLATE_STRING_FORMAT_SEPARATOR, ''), false, fmt);
             }
         } else {
-            if(labels.hasOwnProperty(key + 'Label')) value = labels[key + 'Label'];
+            var keyLabel = key + 'Label';
+            if(labels.hasOwnProperty(keyLabel)) value = labels[keyLabel];
+        }
+
+        if(isOther) {
+            value = ' (' + value + ') ';
```
Comment (archmoj) [inline src/lib/index.js:1107]: This line reminds me of `¯\_(ツ)_/¯`

**Comment by archmoj** [comment]
> Yeah... but now you can't not-have the spacing around it if you don't want it, like if you want to have it at the beginning you'll always have a leading space when it's present.
> 
> I have a slight preference for the way it was before, I was just trying to think of a way to do even better :)

OK. Now with the underscore variants added in https://github.com/plotly/plotly.js/pull/5690/commits/f8c8451aeaf15b986a7dc96fed77247bf6d8bb93, one could achieve desirable behaviour depending on the context.

**Comment by nicolaskruchten** [comment]
Clever! I like it :)

File: src/plots/template_attributes.js [hunk 2]
```diff
@@ -4,15 +4,24 @@ var docs = require('../constants/docs');
...
-    'for details on the date formatting syntax.'
-].join(' ');
+function templateFormatStringDescription(opts) {
+    var supportOther = opts && opts.supportOther;
+
+    return [
+        'Variables are inserted using %{variable},',
+        'for example "y: %{y}"' + (
+            supportOther ?
+                ' as well as %{xother}, {%_xother}, {%_xother_}, {%xother_}, when positions do not match. Please note that underscore could be used before or after *(x|y)other* to provide optional space.' :
```
Comment (alexcjohnson) [inline src/plots/template_attributes.js:14]:
```suggestion
                ' as well as %{xother}, {%_xother}, {%_xother_}, {%xother_}. When showing info for several points, *xother* will be added to those with different x positions from the first point. An underscore before or after *(x|y)other* will add a space on that side, only when this field is shown.' :
```

File: test/jasmine/tests/hover_label_test.js [hunk 3]
```diff
@@ -4831,6 +4831,76 @@ describe('hovermode: (x|y)unified', function() {
         .then(done, done.fail);
     });
 
+    it('should format differing position using *xother* `hovertemplate` and in respect to `xhoverformat`', function(done) {
+        Plotly.newPlot(gd, [{
+            type: 'bar',
+            hovertemplate: '%{y}%{_xother:.2f}',
```
Comment (alexcjohnson) [inline test/jasmine/tests/hover_label_test.js:4837]:
```suggestion
            hovertemplate: 'y(_x):%{y}%{_xother:.2f}',
```
Just to make it easier to read the test output - likewise I'd suggest including the underscores in traces 3 and 4

**Review by alexcjohnson — APPROVED** [review]
💃 Clever indeed, I like it. Just one description comment and a thought about making the tests easier to read, then this is good to go!

**Pull request merged**